src/main/java/net/fabricmc/loom/util/ModProcessor.java

```diff
@@ -58,82 +55,34 @@
-	public static void processMod(File input, File output, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		if (output.exists()) {
-			output.delete();
-		}
+	public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
 
-		remapJar(input, output, project, artifact);
-
-		//Enable this if you want your nested jars to be extracted, this will extract **all** jars
-		if (project.getExtensions().getByType(LoomGradleExtension.class).extractJars) {
-			handleNestedJars(input, project, config, artifact);
+	public static void processMods(Project project, List<ModDependencyInfo> processList) throws IOException {
+		if (processList.isEmpty()) {
+			return;
 		}
 
-		remapaccessWidener(output, project);
-
-		//Always strip the nested jars
-		stripNestedJars(output);
-	}
-
-	private static void handleNestedJars(File input, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		try (JarFile jarFile = new JarFile(input)) {
-			JarEntry modJsonEntry = jarFile.getJarEntry("fabric.mod.json");
-
-			if (modJsonEntry == null) {
-				return;
+		for (ModDependencyInfo info : processList) {
+			if (info.getRemappedOutput().exists()) {
+				info.getRemappedOutput().delete();
 			}
-
-			try (InputStream inputStream = jarFile.getInputStream(modJsonEntry)) {
-				JsonObject json = GSON.fromJson(new InputStreamReader(inputStream), JsonObject.class);
-
-				if (json == null || !json.has("jars")) {
-					return;
-				}
-
-				JsonArray jsonArray = json.getAsJsonArray("jars");
-
-				for (int i = 0; i < jsonArray.size(); i++) {
-					JsonObject jsonObject = jsonArray.get(i).getAsJsonObject();
-					String fileName = jsonObject.get("file").getAsString();
-					project.getLogger().lifecycle(String.format("Found %s nested in %s", fileName, input.getName()));
-					processNestedJar(jarFile, fileName, project, config, artifact);
-				}
-			}
-		}
-	}
-
-	private static void processNestedJar(JarFile parentJar, String fileName, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
-
-		JarEntry entry = parentJar.getJarEntry(fileName);
-
-		if (entry == null) {
-			throw new RuntimeException(String.format("%s was not found in %s", fileName, parentJar.getName()));
-		}
-
-		File nestedFile = new File(extension.getNestedModCache(), fileName.substring(fileName.lastIndexOf("/")));
-
-		try (InputStream jarStream = parentJar.getInputStream(entry)) {
-			FileUtils.copy(jarStream, nestedFile);
 		}
 
-		File remappedFile = new File(extension.getRemappedModCache(), fileName.substring(fileName.lastIndexOf("/")));
+		remapJars(project, processList);
 
-		processMod(nestedFile, remappedFile, project, config, artifact);
+		for (ModDependencyInfo info : processList) {
+			if (!info.getRemappedOutput().exists()) {
+				throw new RuntimeException("Failed to remap mod" + info);
```

Comment:
This won't show why it failed to remap, some error should be thrown earlier that explains why it failed instead of this.

Reply:
This is a catch all, ideally something will throw sooner but there isn’t any more info I can add to this. The log may provide more info.

Reply:
Alright, you're missing a space there then
```suggestion
				throw new RuntimeException("Failed to remap mod " + info);
```